pkg/client/config.go

```diff
@@ -781,6 +783,38 @@ func (ic Intercept) MarshalYAML() (any, error) {
 	return im, nil
 }
 
+type Cluster struct {
+	DefaultManagerNamespace string `json:"defaultManagerNamespace,omitempty" yaml:"defaultManagerNamespace,omitempty"`
+}
+
+// This is used by a different config -- the k8s_config, which needs to be able to tell if it's overridden at a cluster or environment variable level.
+// Hence we don't default to "ambassador" but to empty, so that it can check that no default has been given.
+const defaultDefaultManagerNamespace = ""
+
+func (cc *Cluster) merge(o *Cluster) {
+	if o.DefaultManagerNamespace != defaultDefaultManagerNamespace {
+		cc.DefaultManagerNamespace = o.DefaultManagerNamespace
+	}
+}
+
+var defaultCluster = Cluster{ //nolint:gochecknoglobals // constant
+	DefaultManagerNamespace: defaultDefaultManagerNamespace,
+}
+
+// IsZero controls whether this element will be included in marshalled output.
+func (cc Cluster) IsZero() bool {
+	return cc == defaultCluster
+}
+
+// MarshalYAML is not using pointer receiver here, because Cluster is not pointer in the Config struct.
```

Comment:
Is that the real reason though ? Not sure I get it.

Reply:
I suspect it has to do with serialization, but in truth I copy pasted the comment without being really sure. @thallgren may be able to tell us more